### PR TITLE
MDS-3184 Permit Design Gaps

### PR DESCRIPTION
**Pull request by nayr974**
This PR adds business rules and addresses design gaps in the original Process Permit workflow.

From A/C:

CANNOT  issue the permit if adjusted annual maximum tonnage would jump the application to a separate category

Cannot issue the permit if the adjusted start and end dates would jump the application to a separate category

Issues that need to be resolved - add link to “jump” to that section

Update ‘MERP’ requirement to include the submission MERPS as well as the ones uploaded in CORE (Make this a reminder/not blocking)

Add “CFP” document requirement  (archeological chance find procedure)- (Make this a reminder/not blocking)

When issues are resolved the success screen should read 
”This {permit_type} is ready to be processed and issued.”

If Referral/Consultation/Public Comment has not been started, include a reminder but do not block the issuance. IF the stages are in progress, force them to be completed before issuance. 

“Issuing Inspector” signature is required

Permittee needs to have an address for the permit document - provide link to update party information to update this

### CODE REVIEW

File: services/core-web/common/utils/helpers.js [hunk 1]
```diff
@@ -373,3 +373,82 @@ const getDurationTextOrDefault = (duration, unit) => {
...
+  } else if (proposed < 10000) {
+    isFeeValid = adjusted < 10000;
+  } else if (proposed >= 10000 && proposed < 60000) {
+    isFeeValid = adjusted < 60000;
+  } else if (proposed >= 60000 && proposed < 125000) {
+    isFeeValid = adjusted < 125000;
+  } else if (proposed >= 125000 && proposed < 250000) {
+    isFeeValid = adjusted < 250000;
+  } else {
+    // Anything above 250,000 is valid as the applicatcant alredy paid the max fee.
```
Comment (exzizt) [inline services/core-web/common/utils/helpers.js:417]: applicatcant  --> applicant

File: services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js [hunk 2]
```diff
@@ -349,44 +362,202 @@ export class ProcessPermit extends Component {
...
+      const permittees = this.props.noticeOfWork.contacts.filter(
+        (contact) => contact.mine_party_appt_type_code === "PMT"
+      );
+      if (permittees.length > 1) {
+        validationMessages.push({
+          message:
+            "The application can not have more than one permittee. Verify the correct permittee and remove any others in Contacts under Application.",
+          route: route.NOTICE_OF_WORK_APPLICATION.dynamicRoute(
+            this.props.noticeOfWork.now_application_guid,
+            "application##contacts"
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js:425]: "application##contacts" or "application#contacts"?

File: services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js [hunk 3]
```diff
@@ -349,44 +362,202 @@ export class ProcessPermit extends Component {
...
+    // Mine Emergency Resposne Plan
     if (
       !this.props.noticeOfWork.documents ||
       !this.props.noticeOfWork.documents.some(
         (doc) => doc.now_application_document_type_code === "MRP" && doc.is_final_package
       )
     ) {
       validationMessages.push({
-        message: `The final application package requires a Mine Emergency Response Plan.`,
+        message: `The final application package is missing a Mine Emergency Response Plan. You can Edit the Final Application Package under Administrative.`,
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js:506]: We have inconsistent capitalization for "final application package".

File: services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js [hunk 4]
```diff
@@ -349,44 +362,202 @@ export class ProcessPermit extends Component {
...
+    // Permittee
+    if (
+      this.props.noticeOfWork.contacts &&
+      this.props.noticeOfWork.contacts.some(
+        (contact) => contact.mine_party_appt_type_code === "PMT"
+      )
+    ) {
+      const permittee = this.props.noticeOfWork.contacts.filter(
+        (contact) => contact.mine_party_appt_type_code === "PMT"
+      )[0];
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js:539]: Index out of range error if no permittee has been set?

File: services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js [hunk 5]
```diff
@@ -415,8 +586,10 @@ export class ProcessPermit extends Component {
   );
 
   render = () => {
-    const validationMessages = this.getValidationMessages();
-    const validationErrors = validationMessages.length > 0;
+    const validationErrors = this.getValidationErrors();
+    const validationWarnings = this.getValidationWarnings();
+    const isValidationErrors = validationErrors.length > 0;
```
Comment (exzizt) [inline services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js:591]: hasValidationErrors, hasValidationWarnings?

File: services/core-web/src/styles/generic/layout.scss [hunk 6]
```diff
@@ -291,7 +291,7 @@ body {
   }
 }
 
-.padding-large {
+.padding-lg {
```
Comment (exzizt) [inline services/core-web/src/styles/generic/layout.scss:294]: Can we please change this to "padding-hg"?

**Review by exzizt — DISMISSED** [review]
I've got no major concerns but feel free to address them if you want.

Great work!

File: services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js (hunk 4 above)
Comment (nayr974) [reply services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js:539]: no, there's a .some() conditional above

File: services/core-web/src/styles/generic/layout.scss (hunk 6 above)
Comment (nayr974) [reply services/core-web/src/styles/generic/layout.scss:294]: we can change xxl to hg

**Review by exzizt — DISMISSED** [review]
✔️

**Review by mmumby — APPROVED** [review]
🚀 🚢

**Pull request merged**